api/src/main/java/com/datastrato/gravitino/rel/types/Types.java

```diff
@@ -923,4 +923,12 @@ public int hashCode() {
       return Arrays.hashCode(types);
     }
   }
+
+  /**
+   * @param dataType The data type to check.
+   * @return True if the given data type is allowed to be an auto-increment column.
+   */
+  public static boolean allowAutoIncrement(Type dataType) {
+    return dataType instanceof IntegerType || dataType instanceof LongType;
```

Comment:
Other int types like `smallint` and `tinyint` must also be considered. 

<img width="926" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/3f12a8a2-a6f1-4c14-b5cb-716a43164c83">